src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -506,6 +519,23 @@ bool CartGridStrategy::doProbe(Gcode *gc)
         return false;
     }
 
+    if (do_manual_attach) {
+        // Move to the attachment point defined
+        if (do_home) zprobe->home();
+
+        float x, y, z;
+        std::tie(x, y, z) = m_attach;
+        zprobe->coordinated_move( x, y, z , zprobe->getFastFeedrate());
+
+        gc->stream->printf(" ************************************************************\n");
+        gc->stream->printf("     Ensure probe is attached and trigger probe when done\n");
+        gc->stream->printf(" ************************************************************\n");
+
+        while( !zprobe->getProbeStatus()) {
+            THEKERNEL->call_event(ON_IDLE);
```

Comment:
need to add if(THEKERNEL->is_halted()) return(false);
To avoid hanging the smoothie in the case of no zprobe.